Lib/multiprocessing/shared_memory.py

```diff
@@ -241,6 +242,32 @@ def unlink(self):
             _posixshmem.shm_unlink(self._name)
             unregister(self._name, "shared_memory")
 
+    def rename(self, newname, flags = 0):
+        """Renames a shared memory block.
+
+        The policy how the operation is handled depends on the flag passed.
+        The default behavior is if the newname already exists, it will
+        be unlinked beforehand.
+        With the SHM_RENAME_EXCHANGE flag, the old and new name will
+        be exchanged.
+        With the SHM_RENAME_NOREPLACE flag, an error will be returned
+        if the new name exists.
+        """
+        if platform.system() != "FreeBSD":
+            raise OSError("Unsupported operation on this platform")
+
+        if newname:
```

Comment:
So it simply returns False if newname is empty and returns True otherwise? Why add such check in this method?